### DISCUSSION TITLE
Is initiating a api call using riverpod a good approach inside didChangeDependencies?

### DISCUSSION THREAD

**Discussion by alceil** [rrousselGit/riverpod#3387] [Q&A]
I have a usecase where i need to call the api in the initState of the page but when i do that i get an error. So I put my api call on a didChangeDependencies and it did work. But is it the right approach to do with riverpod.

**Comment by moxiaov587** [comment]
Look at [this](https://riverpod.dev/docs/essentials/do_dont#avoid-initializing-providers-in-a-widget).

**Comment by alceil** [comment]
Okay Thanks @moxiaov587
